libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,270 @@
+        // since we assume dimensions must be positive this is not as issue
+        let width = bitmap
+            .width()
+            .try_into()
+            .map_err(|_foo| ChonkyError::Pdfium)?;
+
+        let height = bitmap
+            .height()
+            .try_into()
+            .map_err(|_foo| ChonkyError::Pdfium)?;
+
+        Ok(match bitmap.format().map_err(|_foo| ChonkyError::Pdfium)? {
+            #[expect(deprecated)]
+            PdfBitmapFormat::BGRA
+            | PdfBitmapFormat::BRGx
+            | PdfBitmapFormat::BGRx
+            | PdfBitmapFormat::BGR => RgbaImage::from_raw(width, height, bytes)
+                .map(DynamicImage::ImageRgba8)
+                .ok_or(ChonkyError::Pdfium)?,
+            PdfBitmapFormat::Gray => GrayImage::from_raw(width, height, bytes)
+                .map(DynamicImage::ImageLuma8)
+                .ok_or(ChonkyError::Pdfium)?,
+        })
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use error_stack::{Report, ResultExt};
+    use insta::assert_binary_snapshot;
+    use pdfium_render::prelude::Pdfium;
+
     use super::*;
 
     #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+    fn pdf_load_success() -> Result<(), Report<ChonkyError>> {
+        let pdfium = Pdfium::default();
+
+        let test_pdf_string = "tests/docs/test-doc.pdf";
+
+        let _pdf = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium)?;
+
+        Ok(())
+    }
+
+    #[test]
+    fn pdf_load_failure() -> Result<(), Report<ChonkyError>> {
+        let pdfium = Pdfium::default();
+
+        let test_pdf_string = "tests/docs/invalid.pdf";
+
+        // Should return an error when loading an invalid PDF
+        let result = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium);
+
+        if result.is_err() {
+            // Expected failure, return Ok
+            Ok(())
+        } else {
+            // Unexpected success, return an error
+            Err(Report::new(ChonkyError::Pdfium).attach_printable("Expected load_pdf to fail"))
+        }
+    }
+
+    #[test]
+    fn pdf_image_conversion() -> Result<(), Report<ChonkyError>> {
+        //create the pdfium instance and bind to static library
+        let pdfium = Pdfium::default();
+
+        let test_pdf_string = "tests/docs/test-doc.pdf";
+
+        let pdf = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium)?;
+
+        //number of pages of pdf
+        let num_pages: usize = pdf.pages().len().into();
+
+        let preprocessed_pdf =
+            pdf_segmentation::pdf_to_images(&pdf).change_context(ChonkyError::Pdfium)?;
+
+        //start by checking if proper amount of images are converted
+        match preprocessed_pdf.len() {
+            x if x == num_pages => (),
```

Comment:
As this is the only match arm you could replace the `match` with `if x != num_pages { return Err(...) }`